[quant] conv_transpose graph patterns

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45078 [quant] conv_transpose graph patterns**

Differential Revision: [D23821580](https://our.internmc.facebook.com/intern/diff/D23821580)

Commit:
[quant] conv_transpose graph patterns

[ghstack-poisoned]

z-a-f commented:
Resubmission of the #43199

vkuzo commented on test/quantization/test_quantize_jit.py:
```diff
@@ -3124,6 +3124,34 @@ def test_conv(self):
+        r"""Compare the result of quantizing conv_transpose layer in
+        eager mode and graph mode
+        """
+        if qengine_is_qnnpack():  # Currently only qnnpack is supported
```
optional nit to save an indent:
```
if not qengine_is_qnnpack():
    return
...
```

vkuzo commented on torch/testing/_internal/common_quantization.py:
```diff
@@ -785,19 +785,25 @@ def forward(self, x, hid):
 
 class ConvModel(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, transposed=False):
```
can we create a new class instead, to follow that `torch.nn.Conv` is a different class than `torch.nn.ConvTranspose`?

vkuzo commented on torch/testing/_internal/common_quantization.py:
```diff
@@ -785,19 +785,25 @@ def forward(self, x, hid):
 
 class AnnotatedConvModel(torch.nn.Module):
-    def __init__(self, qengine):
+    def __init__(self, qengine, transposed=False):
```
same here

vkuzo commented on test/quantization/test_quantize_jit.py:
```diff
@@ -3124,6 +3124,34 @@ def test_conv(self):
+        if qengine_is_qnnpack():  # Currently only qnnpack is supported
+            # eager mode
+            annotated_conv_model = AnnotatedConvModel(
+                torch.backends.quantized.engine, transposed=True).eval()
+            conv_model = ConvModel(transposed=True).eval()
```
creating a new `ConvTransposed` test class might be cleaner

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23821580](https://our.internmc.facebook.com/intern/diff/D23821580)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23821580](https://our.internmc.facebook.com/intern/diff/D23821580)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23821580](https://our.internmc.facebook.com/intern/diff/D23821580)

[ghstack-poisoned]
